--- gdal-svn/ogr/ogrfeature.cpp 2013-04-22 08:38:20.473419728 -0400
+++ gdal-1.7.2/ogr/ogrfeature.cpp       2013-04-25 05:49:08.786169303 -0400
@@ -1201,18 +1136,21 @@
     }
     else if( poFDefn->GetType() == OFTIntegerList )
     {
+        // SD 2010OCT20 - grow buffer if needed
+        char   *szTempBuffer = (char*)CPLMalloc(TEMP_BUFFER_SIZE);
+        int     lenbuffer = TEMP_BUFFER_SIZE;
+
         char    szItem[32];
         int     i, nCount = pauFields[iField].IntegerList.nCount;
 
         snprintf( szTempBuffer, TEMP_BUFFER_SIZE, "(%d:", nCount );
         for( i = 0; i < nCount; i++ )
         {
-            snprintf( szItem, sizeof(szItem), "%d",
-                      pauFields[iField].IntegerList.paList[i] );
-            if( strlen(szTempBuffer) + strlen(szItem) + 6
-                >= sizeof(szTempBuffer) )
+            snprintf( szItem, sizeof(szItem), "%d", pauFields[iField].IntegerList.paList[i] );
+            if( strlen(szTempBuffer) + strlen(szItem) + 6 >= lenbuffer )
             {
-                break;
+                lenbuffer *= 2;
+                szTempBuffer = (char*)CPLRealloc(szTempBuffer, lenbuffer);
             }
             
             if( i > 0 )
@@ -1226,7 +1164,10 @@
         else
             strcat( szTempBuffer, ")" );
         
-        return m_pszTmpFieldValue = CPLStrdup( szTempBuffer );
+        m_pszTmpFieldValue = CPLStrdup( szTempBuffer );
+        CPLFree(szTempBuffer);
+
+        return m_pszTmpFieldValue;
     }
     else if( poFDefn->GetType() == OFTRealList )
     {
@@ -1268,17 +1209,24 @@
     }
     else if( poFDefn->GetType() == OFTStringList )
     {
+        // SD 2013APR03 - grow buffer if needed
+        char   *szTempBuffer = (char*)CPLMalloc(TEMP_BUFFER_SIZE);
+        int     lenbuffer = TEMP_BUFFER_SIZE;
+
+        char    szItem[32];
         int     i, nCount = pauFields[iField].StringList.nCount;
 
         snprintf( szTempBuffer, TEMP_BUFFER_SIZE, "(%d:", nCount );
         for( i = 0; i < nCount; i++ )
         {
             const char  *pszItem = pauFields[iField].StringList.paList[i];
-            
-            if( strlen(szTempBuffer) + strlen(pszItem)  + 6
-                >= sizeof(szTempBuffer) )
+
+            snprintf( szItem, sizeof(szItem), "%s", pszItem );
+           
+            if( strlen(szTempBuffer) + strlen(szItem) + 6 >= lenbuffer )
             {
-                break;
+                lenbuffer *= 2;
+                szTempBuffer = (char*)CPLRealloc(szTempBuffer, lenbuffer);
             }
 
             if( i > 0 )
@@ -1292,7 +1240,10 @@
         else
             strcat( szTempBuffer, ")" );
         
-        return m_pszTmpFieldValue = CPLStrdup( szTempBuffer );
+        m_pszTmpFieldValue = CPLStrdup( szTempBuffer );
+        CPLFree(szTempBuffer);
+
+        return m_pszTmpFieldValue;
     }
     else if( poFDefn->GetType() == OFTBinary )
     {
